code/modules/mob/living/simple_animal/hostile/ordeal/violet.dm

```diff
@@ -209,7 +209,7 @@
 
 /mob/living/simple_animal/hostile/ordeal/violet_midnight/Initialize()
 	. = ..()
-	ability_cooldown = world.time + rand(4 SECONDS, 8 SECONDS)
+	ability_cooldown = world.time + rand(0, ability_cooldown_time)
```

Comment:
I am a bit concerned that this could lead to players getting attacked the instant midnight occurs. Could you keep the minimum at 4?

Reply:
Sure.